File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.cpp
class Solution {
public:
    string sortVowels(string s) {
        string t="";
        for(int i=0;i<s.length();i++){
            if(s[i]=='A'||s[i]=='E' || s[i]=='I'||s[i]=='O' || s[i]=='U'||s[i]=='a' || s[i]=='e'||s[i]=='i' || s[i]=='o'||s[i]=='u')
            t+=s[i];
        }
        sort(t.begin(),t.end());
        int k=0;
        for(int i=0;i<s.length();i++){
            if(s[i]=='A'||s[i]=='E' || s[i]=='I'||s[i]=='O' || s[i]=='U'||s[i]=='a' || s[i]=='e'||s[i]=='i' || s[i]=='o'||s[i]=='u'){
                s[i]=t[k++];
            }

        }
        return s;
    }
};

File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
         String t="";
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='A'||s.charAt(i)=='E' || s.charAt(i)=='I'||s.charAt(i)=='O' || s.charAt(i)=='U'||s.charAt(i)=='a' || s.charAt(i)=='e'||s.charAt(i)=='i' || s.charAt(i)=='o'||s.charAt(i)=='u')
                t+=s.charAt(i);
        }
        char [] chart=t.toCharArray();
        Arrays.sort(chart);
        int k=0;
        StringBuilder ans=new StringBuilder();
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='A'||s.charAt(i)=='E' || s.charAt(i)=='I'||s.charAt(i)=='O' || s.charAt(i)=='U'||s.charAt(i)=='a' || s.charAt(i)=='e'||s.charAt(i)=='i' || s.charAt(i)=='o'||s.charAt(i)=='u'){
                 ans.append(chart[k]);
            k++;
            }
        else ans.append(s.charAt(i));

        }
        return ans.toString();

    }
}